include/logging/log_core.h

```diff
@@ -402,9 +402,12 @@ static inline char z_log_minimal_level_to_char(int level)
 #define LOG_FILTER_FIRST_BACKEND_SLOT_IDX 1
 
 #ifdef CONFIG_LOG_RUNTIME_FILTERING
+#define LOG_CHECK_CTX_LVL_FILTER(ctx, _level, _filter) \
+	(ctx || (_level <= LOG_RUNTIME_FILTER(_filter)))
 #define LOG_RUNTIME_FILTER(_filter) \
 	LOG_FILTER_SLOT_GET(&(_filter)->filters, LOG_FILTER_AGGR_SLOT_IDX)
 #else
+#define LOG_CHECK_CTX_LVL_FILTER(ctx, _level, _filter) (true)
```

Comment:
ctx is not checked in that case?

Reply:
Due to how the filter part resolves into `1` when runtime filters are not used (see previous comment reply), it is disregarded.

If the original code did not intend this then it should likely have been `&&` instead of `||`.